kernel/common/float.rb

```diff
@@ -285,4 +285,32 @@ def floor
     return int if self > 0 or self == int
     return int - 1
   end
+
+  def next_float
+    return NAN if self.nan?
+    return -MAX if self == -INFINITY
+    return INFINITY if self == MAX
+    return Math.ldexp(0.5, MIN_EXP - MANT_DIG + 1) if self.zero?
+
+    frac, exp = Math.frexp self
+
+    if frac == -0.5
+      frac *= 2
+      exp -= 1
+    end
+
+    d = EPSILON / 2
+    d = Math.ldexp(d, MIN_EXP - exp) if exp < MIN_EXP
+
+    result_frac = frac + d
```

Comment:
What does `d` stand for in this case? This is rather picky but I personally despise single letter variables.